components/core/src/clp/NetworkReader.hpp

```diff
@@ -106,7 +107,7 @@ class NetworkReader : public ReaderInterface {
     );
 
     // Destructor
-    virtual ~NetworkReader();
+    ~NetworkReader() override;
```

Comment:
This line isn't related to the topic of this PR, but since this PR is also trying to fix some clang-tidy warnings, I fixed it in passing.